### PR TITLE
Added a login endpoint and user management with peewee

### LINKED ISSUE #17: Login
Add some sort of simple login logic.

### PR DESCRIPTION
**Pull request by ericjperry**
Ripped some IAM stuff from the other project where I could and plopped it in here. Changes include:

* Using peewee to store user credentials, with bcrypt as the password hashing lib.
* Token authorization plugin that expects a token in the Authorization header for requests. Adds the requesting user to handlers that have the 'user' parameter.
* Updated all tests to send an Authorization header.
* Removed the baseservice stuff because I couldn't figure out a succinct way to incorporate that into the other services. I think we should make another backlog item for having a base service of some kind, but for right now just leaving it as-is even though the services repeat some code.

Resolves #17

### CODE REVIEW

File: bakula/services/registration_test.py [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 class RegistrationTest(unittest.TestCase):
+    auth_header = None
+    test_user = None
+
     @classmethod
     def setUpClass(self):
         models.initialize_models({'sqlite_database': ':memory:', 'databaseType': 'sqlite'})
+        iam.create('user', 'some_password')
+        RegistrationTest.test_user = models.User.get(models.User.id == 'user')
+        RegistrationTest.auth_header = {'Authorization': tokenutils.generate_auth_token('password', RegistrationTest.test_user.id, 120)}
```
Comment (KrisSiegel) [inline bakula/services/registration_test.py]: I think this is fine for the demo but in the real release we should use tokens that are randomly generated versus being a serialized secret. Security++

**Comment by KrisSiegel** [comment]
Seems fine. Made a minor comment about security but I don't think it has to be changed for production.

File: bakula/services/registration_test.py (hunk 1 above)
Comment (ericjperry) [inline bakula/services/registration_test.py]: Yeah probably best to have a token store and compare with the existing token to get the user, but I just wanted to make something quick :stuck_out_tongue:

File: bakula/security/iam.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    Args:
+        id: The ID of the user
+        password: The password for the user to verify
+
+    Returns:
+        a boolean if the user is authenticated to use Bakula
+    """
+    try:
+        user = User.get(User.id == id)
+        if bcrypt.hashpw(password, user.password) == user.password:
```
Comment (sapanshah) [inline bakula/security/iam.py]: Please add the following comments to this line:

# Check that an unencrypted password matches one that has
# previously been hashed

**Comment by sapanshah** [comment]
:+1:

**Pull request merged**